### ISSUE TITLE
Accessibility bug:  All features relying on a toast (like the share button) are inaccessible

### ISSUE THREAD

**Issue by SleeplessByte** [bluesky-social/social-app#6322]
### Steps to Reproduce

1. Go to https://bsky.app/
2. Go to any post
3. Click the share post button
4. A toast will appear in the bottom left corner. This toast is **not** announced to any screenreader, and thus, it is as if nothing happens

Before I go into detail, here are people, some of which way smarter than I am, talking about this subject:

- https://adrianroselli.com/2019/06/scraping-burned-toast.html
- https://www.erikkroes.nl/blog/burn-your-toast/
- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html

In short: if the information conveyed in the toast is _necessary_ (and it is right now), then this fails on all fronts. You probably want to ensure the toast is not needed at all, but if you don't want to get rid of it, you want to use a [Live Region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) to announce the information displayed on the screen.

This is what happens in my screen reader:

```text
button share post

*click*

(nothing happens)
``` 

- [ ] Clicking the share button should announce the outcome (content copied to clipboard)

### What platform(s) does this occur on?

Web (Desktop), Web (Mobile)

### Device Info

Not Relevant

### What version of the app are you using?

Web

### Additional Information

I have not tested this on the apps. My guess is that any snackbar or toast that's native to the OS will work (to announce, but PLEASE test this), however most custom implementations do not, so again, please test this.

- [ ] Test a screenreader on iOS to see if the information is announced
- [ ] Test a screenreader on Android to see if the information is announced

**Comment by stubbornella** [comment]
On VoiceOver I don’t hear the copy to clipboard announced after clicking control-option-space 

This is a good fix to make since the rest of the action menu is really well done accessibility-wise and meets the criteria for menus.

**Comment by ashleemboyer** [comment]
Accessibility consultant here, chiming in to +1! I'm seeing this today with VoiceOver. Tiny screen recording here (no sound):
https://bsky.app/profile/ashleemboyer.com/post/3lcxdi5pjp22j
